eras/conway/impl/src/Cardano/Ledger/Conway/Rules/Bbody.hs

```diff
@@ -36,77 +60,265 @@ import Cardano.Ledger.Conway.Rules.Ledgers ()
 
 instance InjectRuleFailure "BBODY" ConwayGovPredFailure (ConwayEra c) where
-  injectFailure = ShelleyInAlonzoBbodyPredFailure . LedgersFailure . injectFailure
+  injectFailure = shelleyToConwayBbodyPredFailure . Shelley.LedgersFailure . injectFailure
+
+shelleyToConwayBbodyPredFailure ::
+  forall era.
+  ShelleyBbodyPredFailure era ->
+  ConwayBbodyPredFailure era
+shelleyToConwayBbodyPredFailure (Shelley.WrongBlockBodySizeBBODY x y) = WrongBlockBodySizeBBODY x y
+shelleyToConwayBbodyPredFailure (Shelley.InvalidBodyHashBBODY x y) = InvalidBodyHashBBODY x y
+shelleyToConwayBbodyPredFailure (Shelley.LedgersFailure x) = LedgersFailure x
+
+alonzoToConwayBbodyPredFailure ::
+  forall era.
+  AlonzoBbodyPredFailure era ->
+  ConwayBbodyPredFailure era
+alonzoToConwayBbodyPredFailure (ShelleyInAlonzoBbodyPredFailure x) = shelleyToConwayBbodyPredFailure x
+alonzoToConwayBbodyPredFailure (Alonzo.TooManyExUnits x y) = TooManyExUnits x y
+
+instance
+  ( DSignable (EraCrypto era) (Hash (EraCrypto era) EraIndependentTxBody)
+  , Embed (EraRule "LEDGERS" era) (EraRule "BBODY" era)
+  , Environment (EraRule "LEDGERS" era) ~ ShelleyLedgersEnv era
+  , State (EraRule "LEDGERS" era) ~ LedgerState era
+  , Signal (EraRule "LEDGERS" era) ~ Seq (AlonzoTx era)
+  , AlonzoEraTxWits era
+  , Tx era ~ AlonzoTx era
+  , Era.TxSeq era ~ AlonzoTxSeq era
+  , Tx era ~ AlonzoTx era
+  , EraSegWits era
+  , AlonzoEraPParams era
+  , InjectRuleFailure "BBODY" AlonzoBbodyPredFailure era
+  , InjectRuleFailure "BBODY" ConwayBbodyPredFailure era
+  , EraRule "BBODY" era ~ ConwayBBODY era
+  , EraTx era
+  , BabbageEraTxBody era
+  ) =>
+  STS (ConwayBBODY era)
+  where
+  type
+    State (ConwayBBODY era) =
+      ShelleyBbodyState era
+
+  type
+    Signal (ConwayBBODY era) =
+      Block (BHeaderView (EraCrypto era)) era
+
+  type Environment (ConwayBBODY era) = BbodyEnv era
+
+  type BaseM (ConwayBBODY era) = ShelleyBase
+
+  type PredicateFailure (ConwayBBODY era) = ConwayBbodyPredFailure era
+  type Event (ConwayBBODY era) = AlonzoBbodyEvent era
+
+  initialRules = []
+  transitionRules = [conwayBbodyTransition @era >> alonzoBbodyTransition @era]
+
+conwayBbodyTransition ::
+  forall era.
+  ( Signal (EraRule "BBODY" era) ~ Block (BHeaderView (EraCrypto era)) era
+  , State (EraRule "BBODY" era) ~ ShelleyBbodyState era
+  , State (EraRule "LEDGERS" era) ~ LedgerState era
+  , Era.TxSeq era ~ AlonzoTxSeq era
+  , Tx era ~ AlonzoTx era
+  , InjectRuleFailure "BBODY" AlonzoBbodyPredFailure era
+  , InjectRuleFailure "BBODY" ConwayBbodyPredFailure era
+  , EraTx era
+  , BabbageEraTxBody era
+  ) =>
+  TransitionRule (EraRule "BBODY" era)
+conwayBbodyTransition = do
+  judgmentContext
+    >>= \( TRC
+            ( _
+              , state@(BbodyState ls _)
+              , UnserialisedBlock _ txsSeq
+              )
+          ) -> do
+        let maxTotalRefScriptSize = 2560 * 1024 :: Int -- 2.5 MB = 2.5 * 1024 * 1024 bytes
```

Comment:
Could this be exposed as a constant for Consensus?

Reply:
The plan is to convert it to a pparam in Dijkstra.  Do you think it would be useful to export it temporarily, or will it cause more overhead/confusion later on?

Reply:
I think it would be nice to use an export from Ledger in Consensus (concretely, in the block forging logic), as it means that you can remove/deprecate it once you implement it as a pparam in Dijkstra, so Consensus will be forced to update.

Reply:
Done, exposed in `Cardano.Ledger.Conway.Rules` as  `maxRefScriptSizePerBlock`  (and same for `maxRefScriptSizePerTx`).